src/main/java/org/junit/rules/TemporaryFolder.java

```diff
@@ -167,16 +167,27 @@ public File newFile() throws IOException {
     }
 
     /**
-     * Returns a new fresh folder with the given name under the temporary
+     * Returns a new fresh folder with the given path under the temporary
      * folder.
      */
     public File newFolder(String folder) throws IOException {
-        return newFolder(new String[]{folder});
+        if (new File(folder).isAbsolute()) {
+            throw new IOException("folder name must be a relative path");
+        }
+        File file = new File(getRoot(), folder);
+        if (!file.mkdirs()) {
+            throw new IOException(
+                    "a folder with the name \'" + folder + "\' already exists");
```

Comment:
Is this error message really accurate? I know it's the same as in the other method but I'm sure there are other reasons why a folder could not be created.